Test/front end unit tests

lposo opened the pull request:
Added tests for the Graph, GraphView, DataCell and NavigationMenu. For most tests, in order to verify if a function was called, a jest spy would be linked to the console log. This spy would check to see if certain messages were called in order to validate the functions. Other tests added include snapshot tests to see if anything changed in regards to the page.

Commit:
Added snapshot tests for frontend.

Added snapshot tests for the GrapoView,, Graph, DataCell. amd App components.

Commit:
Create setupTest.js

Commit:
Added tests for Graph to see if the changing scale functions work.

Added two tests that call the changeXScale/yScale functions. The tests then check if a message was sent to the log.

Commit:
Added tests for GraphViow and Navigation Menu

For GraphView, the tests checks if loading a random dataset works.
For NavigationMenu, the test checks the functionality of opening/closing the drawer.

Commit:
Removed unecessary tests and modified certain test.

Some tests which test if a button existed were already called in other tests, so they were removed. Additionally, to assert if a component exist, the function "toBeTruthy" was used prior. This has been changed to simply be "toEqual(true).

Commit:
Merge branch 'develop' into Test/Front-End-Unit-Tests

Mehrdadmaskull reviewed (commented):
@lposo LGTM, but do we need the snapshot files too?

lposo commented:
@Mehrdadmaskull  They're auto generated when you run the snapshot tests. I guess we can remove them though.

Commit:
Removed Snap files

Mehrdadmaskull reviewed (approved):
LGTM

Commit:
Added tests for GraphView, SearchView, FileUploadView, and Search

Added a test to GraphView to check if the deleteDataset function is called.Added tests for SearchView and FileUploadView that simply check if they were rendered properly.Added test to Search that checks if the forms were rendered properly.

Commit:
fixed wording on test

changed how the GraphView and Graph tests worked.

lposo commented:
Currently, the tests simply verify if components were rendered, then simulate clicking buttons in order to see if the corresponding methods were called.

lposo committed:
Merge branch 'develop' into Test/Front-End-Unit-Tests

lposo committed:
Merge branch 'develop' into Test/Front-End-Unit-Tests

TiraFire commented on Client/src/Components/Graph/Graph.js:
```diff
@@ -22,10 +22,12 @@ export default function Graph(props) {
 
   const handleXScaleClick = () => {
     setXToggle(!isXLog)
+    console.log('Changed X: ' + isXLog)
```
This is a good way to test the change in a function that returns nothing.

TiraFire commented on Client/src/Components/Graph/Graph.js:
```diff
@@ -226,8 +228,8 @@ export default function Graph(props) {
-        <Button onClick={handleYScaleClick} color="primary">Change Y Scale</Button>
+        <Button id='btn1' onClick={handleXScaleClick} color="primary">Change X Scale</Button>
+        <Button id='btn2' onClick={handleYScaleClick} color="primary">Change Y Scale</Button>
       </div>
```
Ids to buttons should be standard, thank you for adding it

TiraFire reviewed (approved):
Looks good, I am happy that there could be some unit tests for the small functions in the front end even if it is just testing the call.

lposo committed:
Merge branch 'develop' into Test/Front-End-Unit-Tests

Commit:
Revert "Merge branch 'develop' into Test/Front-End-Unit-Tests"

This reverts commit 0a65895abe5c30132cbec3510f5f444edac59c5a, reversing
changes made to 2b1f0d2f3e8305011f3422f9810ef40e9809f685.

Commit:
Revert "Revert "Merge branch 'develop' into Test/Front-End-Unit-Tests""

This reverts commit 0074ecc363c6139589c88db28a6b7fe8adc25697.

Commit:
Fixed pack-lock.json format issue
